### PR TITLE
Bump to amtool-0.24.0 (switch from nixpkgs to full rebuild)

### PR DESCRIPTION
**Pull request by ngc104**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -106,9 +106,9 @@ rec {
   terraform_1_0_0 = builtins.trace "terraform_1_0_0 is deprecated use terraform_1" terraform_1;
   terraform_1_0 = builtins.trace "terraform_1_0 is deprecated use terraform_1" terraform_1;
 
-  ansible = pkgs.ansible_2_10;
+  ansible = pkgs.ansible;
```
Comment (eonpatapon) [inline default.nix:109]: No need to change this as nixpkgs is not updated

**Comment by eonpatapon** [comment]
The commits titles are misleading, they don't say what is going on here

**Review by eonpatapon — CHANGES_REQUESTED** [review]

File: default.nix (hunk 1 above)
Comment (ngc104) [reply default.nix]: Removed commit

File: pkgs/amtool.nix [hunk 2]
```diff
@@ -1,21 +1,51 @@
...
+, fetchFromGitHub
+, installShellFiles
+, source
 }:
 
-stdenv.mkDerivation {
+buildGoModule rec {
   pname = "amtool";
-  version = prometheus-alertmanager.version;
+  version = "0.24.0";
```
Comment (eonpatapon) [inline pkgs/amtool.nix:11]:
```suggestion
  version = source.version;
```

File: pkgs/amtool.nix [hunk 3]
```diff
@@ -1,21 +1,51 @@
...
+  rev = "v${version}";
 
-  unpackPhase = ":";
-  installPhase = ''
-    install -m755 -D ${prometheus-alertmanager}/bin/amtool $out/bin/amtool
+  src = fetchFromGitHub {
+    inherit rev;
```
Comment (eonpatapon) [inline pkgs/amtool.nix:15]:
```suggestion
    rev = "v${source.version}";
```

File: pkgs/amtool.nix [hunk 4]
```diff
@@ -1,21 +1,51 @@
...
+, installShellFiles
+, source
 }:
 
-stdenv.mkDerivation {
+buildGoModule rec {
   pname = "amtool";
-  version = prometheus-alertmanager.version;
+  version = "0.24.0";
+  rev = "v${version}";
```
Comment (eonpatapon) [inline pkgs/amtool.nix:12]:
```suggestion
```

File: pkgs/amtool.nix [hunk 5]
```diff
@@ -1,21 +1,51 @@
...
-  installPhase = ''
-    install -m755 -D ${prometheus-alertmanager}/bin/amtool $out/bin/amtool
+  src = fetchFromGitHub {
+    inherit rev;
+    owner = source.owner;
+    repo = source.repo;
+    sha256 = source.sha256;
+  };
+
+  vendorSha256 = "sha256-ePb9qdCTEHHIV6JlbrBlaZjD5APwQuoGloO88W5S7Oc=";
```
Comment (eonpatapon) [inline pkgs/amtool.nix:21]:
```suggestion
  vendorSha256 = source.vendorSha256;
```
And add manually the vendorSha256 in the `source.json`. This is for the autoupdate script.

File: pkgs/amtool.nix [hunk 6]
```diff
@@ -1,21 +1,51 @@
...
+    installShellCompletion amtool.zsh
   '';
 
   meta = with lib; {
-    description = "Alertmanager CLI";
+    description = "Alert dispatcher for the Prometheus monitoring system";
     homepage = "https://github.com/prometheus/alertmanager";
     license = licenses.asl20;
-    maintainers = with maintainers; [ eonpatapon ];
+    maintainers = with maintainers; [ "ngc104" "jpduthilleul" benley fpletz globin Frostman ];
```
Comment (eonpatapon) [inline pkgs/amtool.nix:48]: You can remove upstream maintainers

**Review by eonpatapon — CHANGES_REQUESTED** [review]

File: pkgs/amtool.nix (hunk 5 above)
Comment (ngc104) [reply pkgs/amtool.nix:21]: Fixed

File: pkgs/amtool.nix (hunk 6 above)
Comment (ngc104) [reply pkgs/amtool.nix:48]: Fixed

File: pkgs/amtool.nix (hunk 4 above)
Comment (ngc104) [reply pkgs/amtool.nix:12]: Fixed

File: pkgs/amtool.nix (hunk 3 above)
Comment (ngc104) [reply pkgs/amtool.nix:15]: Fixed

File: pkgs/amtool.nix (hunk 2 above)
Comment (ngc104) [reply pkgs/amtool.nix]: Fixed

**Review by eonpatapon — APPROVED** [review]

**Pull request merged**